crates/ruff_linter/src/rules/pyupgrade/rules/pep695/private_type_parameter.rs

```diff
@@ -133,7 +134,34 @@ pub(crate) fn private_type_parameter(checker: &Checker, binding: &Binding) -> Op
         return Some(diagnostic);
     }
 
-    diagnostic.try_set_fix(|| {
+    diagnostic.try_set_optional_fix(|| {
+        let source = checker.source();
+
+        // Copied from PYI019
+        let reference_is_complex = |id: ResolvedReferenceId| -> bool {
+            let reference = semantic.reference(id);
+            let in_source = &source[reference.range()];
+
+            in_source != old_name
+        };
```

Comment:
I was discussing this issue with @ntBre on Friday -- we think the issue with replacing "complex" stringized references is probably a result of us setting an incorrect range somewhere for `ResolvedReference`s where the reference is a "complex" stringized reference. I'm okay with this for now, but we should try to see if we can fix the underlying bug rather than working around it repeatedly

Reply:
Aha -- it looks like there's already an issue open for this problem. It's here: https://github.com/astral-sh/ruff/issues/10586